canopen/node/local.py

```diff
@@ -34,6 +34,8 @@ def __init__(
         self.add_write_callback(self.nmt.on_write)
         self.emcy = EmcyProducer(0x80 + self.id)
 
+        self.nmt.add_state_change_callback(self._nmt_state_changed)
```

Comment:
I'm wondering whether this should be made opt-in.  There is certainly existing code using `LocalNode` which already manages the TPDO transmissions externally, because the library didn't use to do it.  With this change, these users could end up with duplicated transmissions, or at least things stepping on each others' toes.

Reply:
What would be the preferred way to achieve this? A keyword argument in the `LocalNode` constructor?

Reply:
Yes, I think an additional argument `auto_tpdo: bool = False` would be fine, with appropriate documentation (docstring) of course.

And while we're at it, I think the `_nmt_state_changed` name does not convey what the function does.  It should rather be named `_tpdo_auto_start_stop`, then this line will clearly express what the connection is (NMT state --> TPDO start).

Reply:
> Would the `_tpdo_configuration_write` call inside `set_data` also be opt-in?

No, I think if someone writes to the local node's objects, it is quite expected to affect the PDO settings.  Actually even for RPDOs.  Of course our processing of these parameters is quite rudimentary now and not really standard compliant, but that can be improved step by step.